File: Src/IKayak.Schemas/Models/Booking.cs
namespace IKayak.Schemas.Models
{
   
    public class Booking
    {
        public Booking()
        {
        }

        public Booking(long id, long userId, string tripKey, string kayakKey, string outingDate, string tripDay, string tripTime, string kayakName, KayakType kayakType)
        {
            Id = id;
            UserId = userId;
            TripKey = tripKey;
            KayakKey = kayakKey;
            State = BookingState.Active;
            OutingDate = outingDate;
            Day = tripDay;
            Time = tripTime;
            KayakName = kayakName;
            Type = kayakType;
        }

        public long Id { get; set; }

        public string KayakKey { get; set; }
        
        public KayakType Type { get; set; }

        public string TripKey { get; set; }

        public long UserId { get; set; }

        public BookingState State { get; set; }

        public string OutingDate { get; set; }

        public string Day { get; set; }

        public string Time { get; set; }
        
        public string KayakName { get; set; }
    }
}

File: KayApp/Models/Booking.m
//
//  Booking.m
//  KayApp
//
//  Created by Ilan Levy on 10/11/13.
//  Copyright (c) 2013 Ilan Levy. All rights reserved.
//

#import "Booking.h"

@implementation Booking

@synthesize KayakName;
@synthesize Type;
@synthesize Day;
@synthesize Time;
@synthesize State;
@synthesize TripKey;
@synthesize OutingDate;

-(NSDate*) getAbsoluteDate:(int) minsOffset
{
    NSNumberFormatter * f = [[NSNumberFormatter alloc] init];
    [f setNumberStyle:NSNumberFormatterDecimalStyle];
    NSNumber * myNumber = [f numberFromString:self.OutingDate];
    NSDate *date = [NSDate dateWithTimeIntervalSince1970: myNumber.intValue];
    
    NSCalendar *calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSGregorianCalendar];
    NSDateComponents *components = [calendar components:NSYearCalendarUnit|NSMonthCalendarUnit|NSDayCalendarUnit fromDate:date];
    [components setHour:0];
    [components setMinute:0];
    date = [calendar dateFromComponents:components];
    
    NSArray *list = [self.Time componentsSeparatedByString:@":"];
    int hrs= [f numberFromString:[list objectAtIndex:0]].intValue;
    int mins = [f numberFromString:[list objectAtIndex:1]].intValue;
    int secs = (hrs * 60 + mins) * 60;
    NSDate *absoluteTripDate = [date dateByAddingTimeInterval: secs];
    
    NSDate* absoluteReminderDate = [absoluteTripDate dateByAddingTimeInterval:-minsOffset*60];
    
    return absoluteReminderDate;
}

@end
